src/services/catalog.service.ts

```diff
@@ -79,6 +138,32 @@ export class CatalogService {
 
     return retCategories;
   }
+
+  private getServiceClasses() {
+    let deferred: any = this.$q.defer();
+
+    this.dataService.list({group: 'servicecatalog.k8s.io', resource: 'serviceclasses'}, {}).then((resources: any) => {
```

Comment:
Just return the value from `dataService.list` instead of creating your own deferred.